File: ms-product/src/main/java/com/example/msproduct/entity/Product.java
package com.example.msproduct.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Integer id;
    public String name;
    public String description;
    public Double price;

}

File: frontAngular/src/app/models/product.ts


export interface Product {
  id: number;
  name: string;
  description: string;
  code: string;
  stock: number;
  price: number;
  category: {
    id: number;
    name: string;
  };
}
export interface ProductItemCart {
  product: Product;
  quantity: number;
}
